### ISSUE TITLE
TypeError: ("unsupported operand type(s) for +: 'float' and 'Day'", 'occurred at index 7') under  plot_calibration_purchases_vs_holdout_purchases

### ISSUE THREAD

**Issue by hmikelee** [CamDavidsonPilon/lifetimes#247]
Dear Cameron,
Thank you for great work in lifetimes implementation
I recently encountered TypeError: ("unsupported operand type(s) for +: 'float' and 'Day'", 'occurred at index 7') when I called  plot_calibration_purchases_vs_holdout_purchases
The code ran ok in an older version of python
'3.6.4 |Anaconda, Inc.| (default, Jan 16 2018, 18:10:19) 
[GCC 7.2.0]'
But failed in a newer verion of python
3.6.8 |Anaconda, Inc.| (default, Dec 30 2018, 01:22:34)
[GCC 7.3.0]

Thank you for your help.

Traceback as below
```
TypeError                                 Traceback (most recent call last)
/mnt/efs0/CLV/kpei/Customer-Analytics/qt.py in <module>()
    163                             observation_period_end=last_date_in_trans_data )
    164 bgf.fit(summary_cal_holdout['frequency_cal'], summary_cal_holdout['recency_cal'], summary_cal_holdout['T_cal'])
--> 165 plot_calibration_purchases_vs_holdout_purchases(bgf, summary_cal_holdout, n=10)
    166
    167 plt.savefig('qt_actual_purchases_in_holdout_vs_predicted_purchase.png')

~/anaconda3/lib/python3.6/site-packages/lifetimes/plotting.py in plot_calibration_purchases_vs_holdout_purchases(model, calibration_holdout_matrix, kind, n, **kwargs)
    110     duration_holdout = summary.iloc[0]['duration_holdout']
    111
--> 112     summary['model_predictions'] = summary.apply(lambda r: model.conditional_expected_number_of_purchases_up_to_time(duration_holdout, r['frequency_cal'], r['recency_cal'], r['T_cal']), axis=1)
    113
    114     if kind == "time_since_last_purchase":

~/anaconda3/lib/python3.6/site-packages/pandas/core/frame.py in apply(self, func, axis, broadcast, raw, reduce, result_type, args, **kwds)
   6485                          args=args,
   6486                          kwds=kwds)
-> 6487         return op.get_result()
   6488
   6489     def applymap(self, func):

~/anaconda3/lib/python3.6/site-packages/pandas/core/apply.py in get_result(self)
    149             return self.apply_raw()
    150
--> 151         return self.apply_standard()
    152
    153     def apply_empty_result(self):

~/anaconda3/lib/python3.6/site-packages/pandas/core/apply.py in apply_standard(self)
    255
    256         # compute the result using the series generator
--> 257         self.apply_series_generator()
    258
    259         # wrap results

~/anaconda3/lib/python3.6/site-packages/pandas/core/apply.py in apply_series_generator(self)
    284             try:
    285                 for i, v in enumerate(series_gen):
--> 286                     results[i] = self.f(v)
    287                     keys.append(v.name)
    288             except Exception as e:

~/anaconda3/lib/python3.6/site-packages/lifetimes/plotting.py in <lambda>(r)
    110     duration_holdout = summary.iloc[0]['duration_holdout']
    111
--> 112     summary['model_predictions'] = summary.apply(lambda r: model.conditional_expected_number_of_purchases_up_to_time(duration_holdout, r['frequency_cal'], r['recency_cal'], r['T_cal']), axis=1)
    113
    114     if kind == "time_since_last_purchase":

~/anaconda3/lib/python3.6/site-packages/lifetimes/fitters/beta_geo_fitter.py in conditional_expected_number_of_purchases_up_to_time(self, t, frequency, recency, T)
    198         _b = b + x
    199         _c = a + b + x - 1
--> 200         _z = t / (alpha + T + t)
    201         ln_hyp_term = np.log(hyp2f1(_a, _b, _c, _z))
    202

TypeError: ("unsupported operand type(s) for +: 'float' and 'Day'", 'occurred at index 7')
```

**Comment by CamDavidsonPilon** [comment]
related to #245 #246?

**Comment by hmikelee** [comment]
Dear Cameron,
In my case, summary_data_from_transaction_data function ran ok for both pandas environment. Hence I think it is probably not related to #245. #246.

My problems are in 2 plot routines, plot_calibration_purchases_vs_holdout_purchases and plot_history_alive. 

Plot_history_alive have a different type error:
TypeError: 'numpy.float64' object cannot be interpreted as an integer



My old pandas environment:

In [3]: pd.show_versions(as_json=False)
/home/ec2-user/anaconda3/lib/python3.6/site-packages/psycopg2/__init__.py:144: UserWarning: The psycopg2 wheel package will be renamed from release 2.8; in order to keep installing from binary please use "pip install psycopg2-binary" instead. For details see: <http://initd.org/psycopg/docs/install.html#binary-install-from-pypi>.
  """)

INSTALLED VERSIONS
------------------
commit: None
python: 3.6.4.final.0
python-bits: 64
OS: Linux
OS-release: 4.9.93-41.60.amzn1.x86_64
machine: x86_64
processor: x86_64
byteorder: little
LC_ALL: None
LANG: en_US.UTF-8
LOCALE: en_US.UTF-8

pandas: 0.23.4
pytest: 3.3.2
pip: 19.0.1
setuptools: 39.2.0
Cython: 0.27.3
numpy: 1.15.4
scipy: 1.0.0
pyarrow: None
xarray: None
IPython: 6.4.0
sphinx: 1.6.6
patsy: 0.5.0
dateutil: 2.7.3
pytz: 2018.5
blosc: None
bottleneck: 1.2.1
tables: 3.4.2
numexpr: 2.6.4
feather: None
matplotlib: 2.2.2
openpyxl: 2.4.10
xlrd: 1.1.0
xlwt: 1.3.0
xlsxwriter: 1.0.2
lxml: 4.0.0
bs4: 4.6.0
html5lib: 1.0.1
sqlalchemy: 1.2.9
pymysql: None
psycopg2: 2.7.5 (dt dec pq3 ext lo64)
jinja2: 2.10
s3fs: None
fastparquet: None
pandas_gbq: None
pandas_datareader: None

New Pandas environment:
pd.show_versions(as_json=False)
/home/ec2-user/anaconda3/lib/python3.6/site-packages/psycopg2/__init__.py:144: UserWarning: The psycopg2 wheel package will be renamed from release 2.8; in order to keep installing from binary please use "pip install psycopg2-binary" instead. For details see: <http://initd.org/psycopg/docs/install.html#binary-install-from-pypi>.
  """)

INSTALLED VERSIONS
------------------
commit: None
python: 3.6.8.final.0
python-bits: 64
OS: Linux
OS-release: 4.9.93-41.60.amzn1.x86_64
machine: x86_64
processor: x86_64
byteorder: little
LC_ALL: None
LANG: en_US.UTF-8
LOCALE: en_US.UTF-8

pandas: 0.24.1
pytest: 3.3.2
pip: 19.0.1
setuptools: 39.2.0
Cython: 0.27.3
numpy: 1.14.2
scipy: 1.0.0
pyarrow: None
xarray: None
IPython: 6.4.0
sphinx: 1.6.6
patsy: 0.5.0
dateutil: 2.7.3
pytz: 2018.5
blosc: None
bottleneck: 1.2.1
tables: 3.4.2
numexpr: 2.6.4
feather: None
matplotlib: 2.2.2
openpyxl: 2.4.10
xlrd: 1.1.0
xlwt: 1.3.0
xlsxwriter: 1.0.2
lxml.etree: 4.1.1
bs4: 4.6.0
html5lib: 1.0.1
sqlalchemy: 1.2.9
pymysql: None
psycopg2: 2.7.5 (dt dec pq3 ext lo64)
jinja2: 2.10
s3fs: None
fastparquet: None
pandas_gbq: None
pandas_datareader: None
gcsfs: None

**Comment by CamDavidsonPilon** [comment]
This is fixed in master: 
```
   pip install --upgrade --no-deps git+https://github.com/CamDavidsonPilon/lifetimes.git
```

**Comment by hmikelee** [comment]
Dear Cameron,

Thank you very much. Your fix did fix #247. It seems that there is another one in plot_history_alive. I will open another ticket.

Thank you again for your great help.

hmikelee
